keras_hub/src/models/moonshine/moonshine_encoder.py

```diff
@@ -0,0 +1,142 @@
+from keras import layers
+from keras import models
+from keras import ops
+
+from keras_hub.src.models.moonshine.moonshine_custom_attention import (
+    MHAWithRope,
+)
+from keras_hub.src.models.moonshine.moonshine_custom_feedforward import (
+    FFLinearGelu,
+)
+from keras_hub.src.models.moonshine.moonshine_custom_feedforward import FFSwiGLU
+from keras_hub.src.models.moonshine.moonshine_utils import Arange
+from keras_hub.src.models.moonshine.moonshine_utils import RotaryEmbedding
+
+
+class MoonshineEncoderLayer(layers.Layer):
+    def __init__(self, dim, inner_dim, n_head, ff_mult, ff_swiglu):
+        super().__init__()
+        self.dim = dim
```

Comment:
rename args following naming conventions across other models. Here and everywhere.